data_clean.Rmd

```diff
@@ -0,0 +1,255 @@
+    # Create new snow variable - did day have snow level at any point?
+    snow_heavy_any = snow_heavy_am | snow_heavy_pm,
+    snow_light_any = snow_light_am | snow_light_pm,
+    snow_none_any = snow_none_am | snow_none_pm,
+    snow_unknown_any = snow_unknown_am | snow_unknown_pm,
+    
+    # Decode am_rain
+    rain_heavy_am = grepl("1", am_rain),
+    rain_light_am =  grepl("2", am_rain),
+    rain_none_am =  grepl("3", am_rain),
+    rain_unknown_am =  grepl("4", am_rain),
+    # Decode pm_rain
+    rain_heavy_pm = grepl("1", pm_rain),
+    rain_light_pm =  grepl("2", pm_rain),
+    rain_none_pm =  grepl("3", pm_rain),
+    rain_unknown_pm =  grepl("4", pm_rain),
+    # Create new rain variable - did day have rain level at any point?
+    rain_heavy_any = rain_heavy_am | rain_heavy_pm,
+    rain_light_any = rain_light_am | rain_light_pm,
+    rain_none_any = rain_none_am | rain_none_pm,
+    rain_unknown_any = rain_unknown_am | rain_unknown_pm
+  ) %>%
+  select(-am_snow, -pm_snow, -am_rain, -pm_rain)  # Remove the original, coded columns
+```
+# Subjective Data Cleanup
+The following code chunks involve more subjective transformations and edits than the above code - these are areas where folks are likely to have stronger opinions on what should be done.
+
+## Distances
+Let's convert all distance measurements to kilometers.  Note that distance values with a corresponding `distance_unit` of `NA` are not altered.
+```{r}
+data <- data %>%
+  mutate(
+    # Convert field_distance to kilometers
+    field_distance = if_else(
+      distance_units == "Miles",
+      1.60934 * field_distance,
+      field_distance
+      ),
+    # Convert nocturnal distance to kilometers
+    nocturnal_distance = if_else(
+      distance_units == "Miles",
+      1.60934 * nocturnal_distance,
+      nocturnal_distance
+      ),
+    # Now that all distances are in kilometeres, update `distance_units`, preserving NA values
+    distance_units = if_else(
+      distance_units == "Miles", 
+      "Kilometers",
+      distance_units
+      )
+    )
+```
+
+## Temperature
+Let's look at temperature. I can not definitively say whether the `temp_unit` column actually means anything, so for now we assume all temperatures are in Fahrenheit.
+
+There is a peculiar outlier, where circle `Cedar Key_US-FL` has a min temp of `4464` and a max temp of `NA` for 2012 -- this looks like a case where the `max_temp` was accidentally appended to the end of the min temp.  A range of `44` to `66` degrees seems totally reasonable for December in Florida.
+```{r}
+data <- data %>%
+  mutate(
+    # Hard-code a fix for the 2012 entry of `Cedar Key_US-FL`
+    min_temp = if_else(
+      circle_id == "Cedar Key_US-FL" & count_year == 2012,
+      44,
+      min_temp
+    ),
+    max_temp = if_else(
+      circle_id == "Cedar Key_US-FL" & count_year == 2012,
+      64,
+      max_temp
+    )
+  )
+```
+
+For counts of people, distance, and hours, we fill `NA` values as 0.
```

Comment:
Because we will be using this data from modeling in the Furture, i would prefer to leave NA values as NA because there is a difference between "No people showed up to count birds" and "We dont know if anyone showed up to count birds"